rpc_proxy.go

```diff
@@ -253,13 +263,26 @@ func (p *rpcProxy) Stop() error {
 	return nil
 }
 
+// hasStarted returns true if the rpcProxy has started and is ready to handle
+// requests.
+func (p *rpcProxy) hasStarted() bool {
+	p.mu.RLock()
+	defer p.mu.RUnlock()
+
+	return p.started
+}
+
 // isHandling checks if the specified request is something to be handled by lnd
 // or any of the attached sub daemons. If true is returned, the call was handled
 // by the RPC proxy and the caller MUST NOT handle it again. If false is
 // returned, the request was not handled and the caller MUST handle it.
 func (p *rpcProxy) isHandling(resp http.ResponseWriter,
 	req *http.Request) bool {
 
+	if !p.hasStarted() {
+		return false
```

Comment:
I'm not sure if returning `false` here is a good idea, as the request will just be passed to the next handler. Instead I think we should check whether it's a request we _would_ handle (the same conditions as we check below) and return an error page (perhaps 503 service unavailable) if we did and only `false` if we didn't?